script.php

```diff
@@ -0,0 +1,220 @@
+<?php
+
+$source = $argv[1];
+$type = $argv[2];
+
+$allowedTypes = [
+    'api_operation_map',
+    'model',
+    'general'
+];
+
+if (count($argv) !== 3 || ! in_array($type, $allowedTypes)) {
+    echo 'Usage: php script.php <source> <type>' . PHP_EOL;
+    echo 'Source: filename or url' . PHP_EOL;
+    echo 'Type: use one of available options (api_operation_map, model, general)' . PHP_EOL;
+    exit(1);
+}
+
+$reader = new Reader;
+$data = $reader->getData($source);
+
+$converter = new Converter();
+$converter->convert($data, $type);
+
+class Reader
+{
+    public function getData(string $source): array
+    {
+        // get from url
+        $parsedUrl = parse_url($source);
+        if (isset($parsedUrl['scheme'])) {
+            $curl = curl_init($source);
+            curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
+            $data = curl_exec($curl);
+            if (curl_errno($curl)) {
+                echo 'Error:' . curl_error($curl) . PHP_EOL;
+
+            }
+            curl_close($curl);
+
+            return $this->toArray($data);
+        }
+
+        // get from file
+        if (file_exists($source)) {
+            $data = file_get_contents($source);
+
+            return $this->toArray($data);
+        }
+
+        // Handle other cases or return an error
+        echo 'Check the source!' . PHP_EOL;
+        die();
+    }
+
+    protected function toArray(string $data): array
+    {
+        return json_decode($data, true) ?: yaml_parse($data);
+    }
+}
+
+class Converter
+{
+    const TYPE_API_OPERATION_MAP = 'api_operation_map';
+    const TYPE_MODEL = 'model';
+    const TYPE_GENERAL = 'general';
+
+    public function convert(array $data, string $type): void
+    {
+        match ($type) {
+            static::TYPE_API_OPERATION_MAP => $this->generateApiOperationMap($data),
+            static::TYPE_MODEL => $this->generateModel($data),
+            static::TYPE_GENERAL => $this->generateGeneral($data),
+        };
+    }
+
+    protected function generateApiOperationMap(array $data): void
+    {
+        $paths = $data['paths'] ?? [];
+        $content = 'export const apiOperationMap = {' . PHP_EOL;
+
+        foreach ($paths as $url => $path) {
+            if (! is_array($path)) {
+                continue;
+            }
+
+            foreach ($path as $method => $info) {
+                $operationId = $info['operationId'];
+                if (! $operationId) {
+                    continue;
+                }
+
+                $method = strtoupper($method);
+                $content .= implode(PHP_EOL, [
+                    "    {$operationId}: {",
+                    "        url: \"{$url}\",",
+                    "        method: \"{$method}\"",
+                    '    },' . PHP_EOL,
+                ]);
```

Comment:
add folder "stubs/{operation}/*" on root level and add plain text templates there, which you can replace with strtr(). will be much easier to update (similar to Laravel stubs). In case you have complex generation (like here with loop) - you can just create 2 stubs, and inside one add token like {OPERATIONS_LIST} and then replace with code generated earlier